service/history/queue/processing_queue_collection.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package queue
+
+import (
+	"sort"
+
+	"github.com/uber/cadence/service/history/task"
+)
+
+type (
+	processingQueueCollection struct {
+		level       int
+		queues      []ProcessingQueue
+		activeQueue ProcessingQueue
+	}
+)
+
+// NewProcessingQueueCollection creates a new collection for non-overlapping queues
+func NewProcessingQueueCollection(
+	level int,
+	queues []ProcessingQueue,
+) ProcessingQueueCollection {
+	sortProcessingQueue(queues)
+	queueCollection := &processingQueueCollection{
+		level:  level,
```

Comment:
What does this level mean?

Reply:
Consider it as the collectionID. All the queues inside a queue collection will have the same `level` which equals to the level of the collection (collectionID).